synapse/lib/aha.py

```diff
@@ -897,6 +1026,40 @@ async def addAhaSvc(self, name, info, network=None):
         await self.fire('aha:svcadd', svcinfo=svcinfo)
         await self.fire(f'aha:svcadd:{svcfull}', svcinfo=svcinfo)
 
+    async def reqAhaSvcProxy(self, svcdef, timeout=None):
+
+        proxy = await self.getAhaSvcProxy(svcdef, timeout=timeout)
+        if proxy is not None:
+            return proxy
+
+        mesg = f'The service is not ready {svcfull}.'
+        raise s_exc.NotReady(mesg=mesg)
+
+    async def getAhaSvcProxy(self, svcdef, timeout=None):
+
+        client = await self.getAhaSvcClient(svcdef)
+        if client is None:
+            return None
+
```

Comment:
Added this to prevent an indefinite hang if the client never enters a ready state.
```
        try:
            await asyncio.wait_for(client.ready.wait(), timeout=5.0)
        except asyncio.TimeoutError:
            return None
  ```

Reply:
OBE